### PR TITLE
Be da 0026

### PR DESCRIPTION
**Pull request by ralsalloum**
- company info entity was being added.
- create, update, and get APIs were being added.

### CODE REVIEW

**Review by Kenan-Hussein — APPROVED** [review]
Please wait for Habib review

File: backend/src/Controller/CompanyInfo/CompanyInfoController.php [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+/**
+ * @Route("v1/company/")
+ */
+class CompanyInfoController extends BaseController
+{
+    private AutoMapping $autoMapping;
```
Comment (habibdevsy) [inline backend/src/Controller/CompanyInfo/CompanyInfoController.php:27]: There is no need to define the variable here, it is enough within the constructor

File: backend/src/Controller/CompanyInfo/CompanyInfoController.php [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+     *                  @OA\Property(type="string", property="email"),
+     *                  @OA\Property(type="string", property="bankName"),
+     *                  @OA\Property(type="string", property="stc")
+     *          )
+     *      )
+     * )
+     *
+     * @Security(name="Bearer")
+     */
+    public function create(Request $request): JsonResponse
```
Comment (habibdevsy) [inline backend/src/Controller/CompanyInfo/CompanyInfoController.php:88]: where is validation?

File: backend/src/Manager/CompanyInfo/CompanyInfoManager.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+        if (! $companyInfoAll) {
+            $companyInfoEntity = $this->autoMapping->map(CompanyInfoCreateRequest::class, CompanyInfoEntity::class, $request);
+
+            $this->entityManager->persist($companyInfoEntity);
+            $this->entityManager->flush();
+
+            return $companyInfoEntity;
+        }
+        else {
```
Comment (habibdevsy) [inline backend/src/Manager/CompanyInfo/CompanyInfoManager.php]: no need for (else) here:
 if (! $companyInfoAll) {
          ...
            return $companyInfoEntity;
        }
return CompanyInfoConstant::COMPANY_INFO_EXISTS;

File: backend/src/Service/CompanyInfo/CompanyInfoService.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    public function create(CompanyInfoCreateRequest $request)
+    {
+        $companyInfoResult = $this->companyInfoManager->create($request);
+
+        if ($companyInfoResult === CompanyInfoConstant::COMPANY_INFO_EXISTS) {
+            return $this->getCompanyInfo();
+
+        } else {
```
Comment (habibdevsy) [inline backend/src/Service/CompanyInfo/CompanyInfoService.php]: no need for (else) here too.

**Review by habibdevsy — APPROVED** [review]

File: backend/src/Controller/CompanyInfo/CompanyInfoController.php (hunk 2 above)
Comment (ralsalloum) [reply backend/src/Controller/CompanyInfo/CompanyInfoController.php:88]: Validation for which field? I mean the admin may enter values for specific fields, not for all.

**Pull request merged**